src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -1339,6 +1193,9 @@ private static bool IsValidCallback(MethodInfo method, ParameterInfo[] parameter
                 [RequiresUnreferencedCode(DataContract.SerializerTrimmerWarning)]
                 get
                 {
+                    // NOTE TODO smolloy - Oddly, this shortcut was not here in NetFx.  I'm not sure this was added strictly as a perf-optimization though.
+                    // See https://github.com/dotnet/runtime/commit/5d2dba7ffcc4ad6a11a570f5bd8d1964000099ea
+                    // This shouldn't change between invocations though, right? So seems like a good thing to leave in.
                     if (_knownDataContracts != null)
```

Comment:
There is a subtle change in behavior. On NetFx, using the setter to set the KnownDataContracts when you haven't ever called the getter before and then using the getter will clobber what you set as _isKnownTypeAttributeChecked is false. In Core, setting the property will always return what you set due to this check.

Reply:
Yeah, I see that. Actually, looking at the commit that added this, it seems like Shin was just trying to add this as a simple get/set property for NetNative, but keep the old behavior otherwise. Then the #ifdefs went away and we got this funkiness. This shortcut should probably be removed, eh?